src/util/exception.cpp

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <util/exception.h>
+
+#include <logging.h>
+#include <tinyformat.h>
+
+#include <iostream>
+#include <typeinfo>
+
+#ifdef WIN32
+#include <compat/compat.h>
```

Comment:
In f12cd1d77bc062f1313f98a8aaea8afd96c4f52e:
Why is compat being included here? if it's for a `windows.h` include (which isn't in compat anyways), then I'd rather we just include `windows.h`.